client/state/data-layer/wpcom-http/utils.js

```diff
@@ -59,12 +62,12 @@ export const getProgress = action => get( action, 'meta.dataLayer.progress', nul
  *   onProgress :: ReduxStore -> Action -> Dispatcher -> ProgressData
  *
  * @param {Function} initiator called if action lacks response meta; should create HTTP request
- * @param {Function} onSuccess called if the action meta includes response data
- * @param {Function} onError called if the action meta includes error data
+ * @param {Function} [onSuccess] called if the action meta includes response data
+ * @param {Function} [onError] called if the action meta includes error data
  * @param {Function} [onProgress] called on progress events when uploading
  * @returns {?*} please ignore return values, they are undefined
  */
-export const dispatchRequest = ( initiator, onSuccess, onError, onProgress = null ) => ( store, action, next ) => {
+export const dispatchRequest = ( initiator, onSuccess = noop, onError = noop, onProgress = noop ) => ( store, action, next ) => {
```

Comment:
actually I'd strongly prefer no to make `onSuccess` and `onError` optional. I'm fine if people want to explicitly state that they are opting out, but the very essence of a network request is a task that can succeed or fail. if we silently drop it then we're absolutely not thinking about the consequences of our actions :wink: and so I'd rather leave those in as top-level mandates in order to encourage thinking about them (primarily of thinking about failure)

could we leave this just as an optional progress handler?